Makefile

```diff
@@ -8,6 +18,17 @@ else
 	CARGO_COMPILE_OPTS = --release
 endif
 
+BUILD_DATE=$(shell date -u +'%Y%m%dT%H%M%SZ')
```

Comment:
We should modify the format string to be RFC 3339 as defined in the label schema: `%Y-%m-%dT%H:%M:%SZ`

Reply:
@jakedipity so i'm playing with this over in the tooling-elasticsearch repository.

RFC3339 doesnt work inside a docker tag because of the colons, and on the next few line we make the build tag using this.

What I've landed on is:

```
BUILD_DATE=$(shell date -u "'%Y-%m-%d")
BUILD_TIME=$(shell date -u "%H:%M:%SZ")
BUILD_CREATED=$(BUILD_DATE)T$(BUILD_TIME)
BUILD_TAG=$(BUILD_DATE).$(BUILD_VERSION)
```
And then using `BUILD_CREATED` in the Dockerfile, which aligns it with teh new OCI tag better.